### PR TITLE
Topic/operators

### PR DESCRIPTION
**Pull request by telephon**
The remaining tests have succeeded, I'm quite confident now that this implementation is a good choice.

### CODE REVIEW

File: include/plugin_interface/SC_InlineBinaryOp.h [hunk 1]
```diff
@@ -337,27 +337,97 @@ inline int sc_fold(int in, int lo, int hi)
...
+	return (float) sc_gcd((long) u, (long) v);
+}
+
+
+inline float sc_lcm(float u, float v)
+{
+    return (float) sc_lcm((long) u, (long) v);
+}
+
+
```
Comment (telephon) [inline include/plugin_interface/SC_InlineBinaryOp.h]: actually, thinking it through, ````doubles```` should be treated just as UGen floats, so you are right, let's add them. For rational lcm/gcd, a different method would be in place anyway.

File: include/plugin_interface/SC_InlineBinaryOp.h [hunk 2]
```diff
@@ -337,27 +337,97 @@ inline int sc_fold(int in, int lo, int hi)
...
+    else
+        return (a * b) / sc_gcd(a, b);
+}
+
+
+
+inline float sc_gcd(float u, float v)
+{
+	return (float) sc_gcd((long) u, (long) v);
+}
```
Comment (telephon) [inline include/plugin_interface/SC_InlineBinaryOp.h]: should truncating be done by a method or is the cast to (long) sufficient?

Comment (timblechmann) [inline include/plugin_interface/SC_InlineBinaryOp.h]: sufficient as in "works" or sufficient as in "readable"?

personally, i'd make this explicit and use std::trunc [1], then interested readers of the code will know that it is done on purpose

[1] http://en.cppreference.com/w/cpp/numeric/math/trunc

Comment (telephon) [inline include/plugin_interface/SC_InlineBinaryOp.h]: Yes, explicit is much better, I'll change it.

**Pull request merged**